### PR TITLE
Copy gen_snapshots using python's shutil.copy, avoid links

### PR DESCRIPTION
**Pull request by aam**
Default implementation of copy does it via hardlink, which seems to be causing issues with Gatekeeper on mac.

BUG=https://github.com/flutter/flutter/issues/154437

### CODE REVIEW

**Review by cbracken — APPROVED** [review]
lgtm. Alternatively, you could tweak the macOS copy rule in the buildroot to drop the ln and just always rsync.

**Review by mraleph — APPROVED** [review]

**Comment by chinmaygarde** [comment]
May I land this? Seems good to go.

**Comment by aam** [comment]
> May I land this? Seems good to go.

I wanted to get another failure before landing this to confirm that this will fix it. We haven't had a failure in a few days.

File: lib/snapshot/BUILD.gn [hunk 1]
```diff
@@ -198,12 +198,18 @@ if (host_os == "mac" && (target_os == "mac" || target_os == "ios")) {
     gen_snapshot_target =
         "$dart_src/runtime/bin:$gen_snapshot_target_name($build_toolchain)"
 
-    copy(target_name) {
+    action(target_name) {
```
Comment (zanderso) [inline lib/snapshot/BUILD.gn:201]: What's the rationale for only using it here and not updating the toolchain definition so that all copy targets use it?

Comment (aam) [reply lib/snapshot/BUILD.gn:201]: It was to limit impact to only `gen_snapshot`s. Only them had an issue with Gatekeeper.
Generally speaking `ln` has an advantage of reduced disk space usage.

Comment (zanderso) [reply lib/snapshot/BUILD.gn:201]: Presumably, Gatekeeper would also impact other binaries as well. If this is the fix for `gen_snapshot`, then we wouldn't want to go through all this investigation again a second time for some other binary. I think it's fine to do it just for `gen_snpashot` for now as an experiment, but if it works, there has to be follow-up work to deploy the solution more generally.

**Comment by chinmaygarde** [comment]
Triage: The investigation of this issue might have gone in a different direction. Do we still need this PR?

**Comment by aam** [comment]
> Triage: The investigation of this issue might have gone in a different direction. Do we still need this PR?

Have been waiting for a recent failure before submitting this fix.
The failure just happened https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8733431533211634625/+/u/build_ci_ios_debug_sim_arm64_extension_safe/stdout
So submitting now.

**Comment by chinmaygarde** [comment]
I ran into the following error after pulling past this commit and building today. https://github.com/flutter/flutter/issues/157367

**Comment by chinmaygarde** [comment]
Reason for revert: Incremental builds are broken. https://github.com/flutter/flutter/issues/157367

**Pull request merged**